### ISSUE TITLE
IOS send SMS action bugged

### ISSUE THREAD

**Issue by squidsareevil** [FlutterFlow/flutterflow-issues#694]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

When I use the send SMS action for IOS. Imessage is not able to parse the information sent by flutterflow correctly, and puts the phone number AND body message in the phone section. You can see the number followed by ?body=. I believe this is a parsing issue from Flutterflow's side. The same send SMS action in Android works properly.

### Expected Behavior

What I put in the recipient section should go to phone. What I put in text should go in the body of the text. 

### Steps To Reproduce

1. Create a send action
2. Input phone number to Phone number section
3. Input text to SMS body section
4. Run app and trigger action on IOS
5. Notice that all content went to phone number only

### Bug Report Code

ITEezsj104lPn7wG0KncavloqwMwGkQ5a7sviu5+FBQaI+KrE7MUfuP7dkloWOqHYWNqG1SGnnMCw6nlhvHhOO5dCy2VfrpEybhPUxDiRmSXRLGaCLmCR0RtM/lXGBGj0cORmiMnPvhhc1IS1F2qGu+Ab3qCf9qOYwx5e6fDbOY=

### Context

I am trying to use SMS to send an invite link to my project. It does not work on IOS

### Additional info

![bug 2](https://user-images.githubusercontent.com/73563502/228933599-721cd070-ba13-4c2a-890f-7424fd86b168.jpg)
![bug1](https://user-images.githubusercontent.com/73563502/228933614-bc66ee09-2b8f-422e-9401-d21ace2bcd06.jpeg)

### Environment

```markdown
- FlutterFlow version: V3.1
- Platform: Web
- Browser name and version: Edge
- Operating system and version affected: IOS
```

**Comment by hariprasadms** [comment]
Hi @squidsareevil - Thanks for reporting the issue. I can re-produce the issue. 

![image](https://user-images.githubusercontent.com/17435678/229097158-b267310a-7a1c-4b5c-bbd4-cbb9f7cd1655.png)

**Comment by squidsareevil** [comment]
A similar issue happens when trying to open apple maps while plugging in an address through the open maps action. The IOS forward is not able to read the address as an address and instead defaults to 0,0 LTG, LNG. This problem does not happen in Android/google maps. It is able to read the address field from flutterflow correctly and locate to the address. 

Both of these problems seem like miss parsing of the data flutterflow forwards. Please help asap as they are essential to my app functionality! Thanks. 

![339750494_236323972242213_7778463951518678041_n](https://user-images.githubusercontent.com/73563502/230667576-2acd5794-fdac-44d7-b5ae-4a12b272c677.jpg)

**Comment by wenkaifan0720** [comment]
Hi @squidsareevil. Sorry for the late reply. I have identified that this is because Apple decides to use a different scheme compared to other platforms for forming the SMS action, and the package we used didn't properly handle it. I have made a special fix for it. As for the launch map action, from what I just tested, it seems to be working fine on my iPhone SE 2020, running iOS 16.4. You do need to grant the app with location access if you are trying to use the user's current location as the initial location.
